gazebo_state_plugins/src/GazeboObjectInfo.cpp

```diff
@@ -194,15 +249,30 @@ GazeboObjectInfo::ObjectMsg GazeboObjectInfo::createBoundingBoxObject(physics::M
         for (cit=colls.begin(); cit!=colls.end(); ++cit)
         {
             physics::CollisionPtr c=*cit;
-            
-            math::Pose rel_pose=c->GetRelativePose();
+
+#if GAZEBO_MAJOR_VERSION >= 8
+            gz_math_Pose3d rel_pose=c->RelativePose();
+#else
+            gz_math_Pose3d rel_pose=c->GetRelativePose();
+#endif
 
             //ROS_INFO("Collision for model %s: %s, pos %f %f %f",model->GetName().c_str(),link->GetName().c_str(),rel_pose.pos.x,rel_pose.pos.y,rel_pose.pos.z);
             //math::Pose w_pose=c->GetWorldPose();
             //ROS_INFO("World pos for model %s: %s, pos %f %f %f",model->GetName().c_str(),link->GetName().c_str(),w_pose.pos.x,w_pose.pos.y,w_pose.pos.z);
 
-            math::Pose coll_pose=rel_pose+link_pose; //XXX c->GetWorldPose() does not work for non-static objects, so it has to be relative pose and link world pose
+            gz_math_Pose3d coll_pose=rel_pose+link_pose; //XXX c->GetWorldPose() does not work for non-static objects, so it has to be relative pose and link world pose
             geometry_msgs::Pose pose;
+
+#if GAZEBO_MAJOR_VERSION >= 8
+
```

Comment:
Similar to the comment above, here it would be nice to have a helper function
```cpp
geometry_msgs/Pose getPose(const GzPose3d &pose);
```
then no `#if` for the version is required here.